src/core/featuremodel.cpp

```diff
@@ -323,7 +323,11 @@ bool FeatureModel::setData( const QModelIndex &index, const QVariant &value, int
 
       bool success = mFeature.setAttribute( index.row(), val );
       if ( success )
+      {
         emit dataChanged( index, index, QVector<int>() << role );
+        // emit a feature changed signal so the attribute form's currentFeature has up-to-date values
+        emit featureChanged();
```

Comment:
PSA: this change is needed to insure QML code relying on a feature model's feature property gets served a feature with up-to-date value (e.g. https://github.com/opengisch/QField/blob/master/src/qml/editorwidgets/ExternalResource.qml#L104 ). I'm slightly anxious at this change, but I think it's the right thing to do here.